File: 125.valid-palindrome.py
# @leet start
class Solution:
    def isPalindrome(self, s: str) -> bool:
        s="".join(char.lower() for char in s if char.isalnum())
        l=0
        r=len(s)-1
        while l<r:
            if s[r]!=s[l]:
                return False
            l+=1
            r-=1
        return True
        
# @leet end

File: 125.valid-palindrome.ts
// @leet start
function isPalindrome(s: string): boolean {
	s = s.replace(/[^A-Za-z0-9]/g, "").toLowerCase();
	let l = 0;
	let r = s.length - 1;
	while (l < r) {
		if (s.charAt(l) !== s.charAt(r)) return false;
		l++;
		r--;
	}
	return true;
}
// @leet end
